desktop/cmp/persistenceManager/PersistenceManagerModel.ts

```diff
@@ -0,0 +1,400 @@
+                    text: this.getHierarchyDisplayName(it.name),
+                    selected: this.selectedId === it.id,
+                    id: it.id,
+                    isFavorite: true
+                }))
+            );
+        }
+
+        sortedGroupKeys.forEach(group => {
+            ret.push({itemType: 'divider', text: group});
+            ret.push(...this.hierarchicalItemSpecs(sortBy(groupedViews[group], 'name')));
+        });
+        return ret;
+    }
+
+    get persistState() {
+        const ret: PlainObject = {};
+        if (this.selectedId) ret.selectedId = this.selectedId;
+        if (this.favorites) ret.favorites = this.favorites;
+        return ret;
+    }
+
+    // Internal persistence provider, used to save *this* model's state, i.e. selectedId
+    private readonly _provider;
+
+    constructor({
+        entity,
+        omitDefaultMenuComponent,
+        onChangeAsync,
+        persistWith,
+        canManageGlobal,
+        enableTopLevelSaveButton = true
+    }: PersistenceManagerConfig<T>) {
+        super();
+        makeObservable(this);
+
+        this.entity = this.parseEntity(entity);
+        this._canManageGlobal = canManageGlobal;
+        this.enableTopLevelSaveButton = enableTopLevelSaveButton;
+        this.omitDefaultMenuComponent = omitDefaultMenuComponent;
+        this.onChangeAsync = onChangeAsync;
+        this.saveDialogModel = new SaveDialogModel(this, this.entity.name);
+        this.manageDialogModel = new ManageDialogModel(this);
+        // Set up internal PersistenceProvider -- fail gently
+        if (persistWith) {
+            try {
+                this._provider = PersistenceProvider.create({
+                    path: 'persistenceManager',
+                    ...persistWith
+                });
+
+                const state = this._provider.read();
+                if (state?.selectedId) this.selectedId = state.selectedId;
+                if (state?.favorites) this.favorites = state.favorites;
+                this.addReaction({
+                    track: () => this.persistState,
+                    run: state => this._provider.write(state)
+                });
+            } catch (e) {
+                this.logError('Error applying persistWith', persistWith, e);
+                XH.safeDestroy(this._provider);
+                this._provider = null;
+            }
+        }
+
+        this.loadAsync();
+    }
+
+    // TODO - Carefully review if this method needs isStale checks, and how to properly implement them.
+    override async doLoadAsync(loadSpec: LoadSpec) {
+        const {name} = this.entity,
+            rawViews = await XH.jsonBlobService.listAsync({type: name, includeValue: true});
+
+        runInAction(() => (this.views = this.processRaw(rawViews)));
+
+        // Always call selectAsync to ensure pendingValue updated and onChangeAsync callback fired if needed
+        const id = this.selectedView?.id ?? this.views[0].id;
```

Comment:
Worth discussing - do we want to fallback on the first view like this? What about for an app that allows us to not be in a view?